folium/features.py

```diff
@@ -409,9 +411,13 @@ class GeoJson(Layer):
                     e.target.setStyle({{ this.get_name() }}_highlighter(e.target.feature));
                 },
                 {%- endif %}
+                {%- if this.zoom_on_click %}
                 click: function(e) {
-                    {{ this.parent_map.get_name() }}.fitBounds(e.target.getBounds());
+                    if (typeof e.target.getBounds === 'function') {
+                        {{ this.parent_map.get_name() }}.fitBounds(e.target.getBounds());
+                    }
```

Comment:
Here, how about to handle zooming for point geometries as well? I'm not sure about best generalized options for the `.fitBounds()` [method](https://leafletjs.com/reference-1.6.0.html#fitbounds-options), it supports both `padding` and `maxZoom` options.
```
else {
    const bounds = L.latLngBounds([e.target.getLatLng()])
    {{ this.parent_map.get_name() }}.fitBounds(bounds, {padding: [5,5]});
}
```

Reply:
I think your code is fine but I'd rather not introduce more fragility here. This may work for point geometries but I'm unsure what new bugs it might introduce. I'll just merge the PR as is, let's consider any new functionality separately.